### ISSUE TITLE
argument of .zip()

### ISSUE THREAD

**Issue by kasugaiasuka** [mihaifm/linq#56]
IEnumerable&lt;T&gt;.zip() accepts U[] (or IEnumerable&lt;U&gt;) as the first argument.

```javascript
Enumerable.from([1, 2, 3])
    .zip(["a", "b", "c"], (first, second) => `${first}${second}`)
    .toArray();

// => ["1a", "2b", "3c"]
```

So, definitions should be like this:

```javascript
zip<U, TResult>(second: IEnumerable<U>, resultSelector: (first: T, second: U, index: number) => TResult): IEnumerable<TResult>;
zip<U, TResult>(second: { length: number;[x: number]: U; }, resultSelector: (first: T, second: U, index: number) => TResult): IEnumerable<TResult>;
zip<U, TResult>(second: U[], resultSelector: (first: T, second: U, index: number) => TResult): IEnumerable<TResult>;
```

**Comment by mihaifm** [comment]
This seems to be correct, I will make the changes.

**Comment by mihaifm** [comment]
Resolved (b815a82950ed484bcddfed1e0b4c132f84a76b1a), thanks for your help.
